### PR TITLE
add property and version modules

### PR DESCRIPTION
**Pull request by rahulchaphalkar**
- Add property enum
- Add version module

### CODE REVIEW

File: crates/openvino/src/property.rs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+impl AsRef<str> for PropertyKey {
+    fn as_ref(&self) -> &str {
+        match self {
+            PropertyKey::SupportedProperties => "SUPPORTED_PROPERTIES",
+            PropertyKey::AvailableDevices => "AVAILABLE_DEVICES",
+            PropertyKey::OptimalNumberOfInferRequests => "OPTIMAL_NUMBER_OF_INFER_REQUESTS",
+            PropertyKey::RangeForAsyncInferRequests => "RANGE_FOR_ASYNC_INFER_REQUESTS",
+            PropertyKey::RangeForStreams => "RANGE_FOR_STREAMS",
+            PropertyKey::DeviceFullName => "FULL_DEVICE_NAME",
+            PropertyKey::DeviceCapabilities => "OPTIMIZATION_CAPABILITIES",
```
Comment (abrown) [inline crates/openvino/src/property.rs:100]:
```suggestion
            PropertyKey::DeviceCapabilities => "DEVICE_CAPABILITIES",
```

Typo?

File: crates/openvino/src/property.rs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            RwPropertyKey::CacheMode => "CACHE_MODE",
+            RwPropertyKey::NumStreams => "NUM_STREAMS",
+            RwPropertyKey::Affinity => "AFFINITY",
+            RwPropertyKey::InferenceNumThreads => "INFERENCE_NUM_THREADS",
+            RwPropertyKey::HintEnableCpuPinning => "ENABLE_CPU_PINNING",
+            RwPropertyKey::HintEnableHyperThreading => "ENABLE_HYPER_THREADING",
+            RwPropertyKey::HintPerformanceMode => "PERFORMANCE_HINT",
+            RwPropertyKey::HintSchedulingCoreType => "SCHEDULING_CORE_TYPE",
+            RwPropertyKey::HintInferencePrecision => "INFERENCE_PRECISION_HINT",
+            RwPropertyKey::HintNumRequests => "PERFORMANCE_HINT_NUM_REQUESTS",
```
Comment (abrown) [inline crates/openvino/src/property.rs:123]: Same here?

**Review by abrown — APPROVED** [review]
It occurs to me that we should link to the documentation for these properties in case someone is interested in knowing what they actually do.

File: crates/openvino/src/property.rs (hunk 1 above)
Comment (rahulchaphalkar) [reply crates/openvino/src/property.rs:101]: This is consistent with [property keys](https://github.com/openvinotoolkit/openvino/blob/34caeefd07800b59065345d651949efbe8ab6649/src/bindings/c/src/ov_property.cpp#L15C13-L15C48) where these naming discrepancies between `device`<->`optimization` , `model`<->`network` exist

**Pull request merged**